File: src/main/java/me/sul/worldmanager/summonmob/mobtype/AutoSummonableMob.java
package me.sul.worldmanager.summonmob.mobtype;

import org.bukkit.Location;
import org.bukkit.entity.Entity;

public interface AutoSummonableMob {
    Entity summonMob(Location loc);
    double getSpawnChance();
    int getMinDistance();
    int getMaxDistance();
}

File: src/main/kotlin/kr/sul/worldmanager/summonmob/mobtype/AutoSummonableMob.kt
package kr.sul.worldmanager.summonmob.mobtype

import org.bukkit.Location
import org.bukkit.entity.Entity

interface AutoSummonableMob {
    fun summonMob(loc: Location): Entity
    val spawnChance: Double
    val minDistance: Int
    val maxDistance: Int
}